core/src/main/java/com/graphhopper/reader/dem/USGSProvider.java

```diff
@@ -41,21 +41,56 @@ boolean isOutsideSupportedArea(double lat, double lon) {
 
     @Override
     double getMinLatForTile(double lat) {
-        return 37.75;
+        return (double) Math.floor(lat * 4) / 4;
     }
 
     @Override
     double getMinLonForTile(double lon) {
-        return -122.5;
+        return (double) Math.floor(lon * 4) / 4;
     }
 
     @Override
     String getFileNameOfLocalFile(double lat, double lon) {
+        // TODO(steveliu): Implement.
         return filename + ".tif";
     }
 
+    /**
+     * The USGS National Elevation Dataset (NED)'s 1/9th arc-second DEM offering
+     * categorizes individual 0.25x0.25 degree tiles using the northwestern
+     * corner of each tile. For example, ned19_n37x75_w122x50 means that the
+     * corners of the tile are (starting from the northwestern corner and moving
+     * clockwise):
+     * <ul>
+     *   <li>37.75, -122.50</li>
+     *   <li>37.75, -122.25</li>
+     *   <li>37.50, -122.25</li>
+     *   <li>37.50, -122.50</li>
+     * </ul>
+     * @param lat latitude in degrees, ranges from [-90.0, 90.0]
+     * @param lon longitude in degrees, ranges from [-180.0, 180.0]
+     * @return
+     */
     @Override
     String getFileName(double lat, double lon) {
+        filename = "ned19_";
+
+        double latAdjusted = Math.abs(Math.ceil(lat * 4) / 4);
+        String latString = String.valueOf(latAdjusted);
+        int indLat = latString.indexOf('.');
+        if(latString.length() < indLat + 3) {
+            latString += "0";
+        }
```

Comment:
This looks a little dicey. Instead of converting the `double` to `String`, just check if the latitude or longitude is greater than `10` or `100` and zero-pad accordingly.